hrjobcontract/CRM/Hrjobcontract/Form/PayScale.php

```diff
@@ -133,35 +153,62 @@ static function formRule($fields, $files, $self) {
   public function postProcess() {
     if ($this->_action & CRM_Core_Action::DELETE) {
       CRM_Hrjobcontract_BAO_PayScale::del($this->_id);
-      CRM_Core_Session::setStatus(ts('Selected pay scale has been deleted.'), 'Success', 'success');
+      $deleteMsg = ts('Selected pay scale has been deleted.');
+      Session::setStatus($deleteMsg, 'Success', 'success');
     }
     else {
-      $params = $ids = array( );
-      // store the submitted values in an array
-      $params = $this->exportValues();
-
-      foreach (array('pay_scale', 'currency', 'amount', 'pay_frequency') as $key => $index) {
-        if (!array_key_exists($index, $params)) {
-          $params[$index] = 0;
-        }
-      }
+      $params = $this->getParams();
 
       if ($this->_action & CRM_Core_Action::UPDATE) {
         $params['id'] = $this->_id;
       }
 
       $payScale = CRM_Hrjobcontract_BAO_PayScale::create($params);
+      $payScaleName = $payScale->pay_scale;
 
       if ($this->_action & CRM_Core_Action::UPDATE) {
-        CRM_Core_Session::setStatus(ts('The Pay Scale for \'%1\' has been updated.', array( 1 => $payScale->pay_scale)), 'Success', 'success');
+        $msg = ts(
+          'The Pay Scale for \'%1\' has been updated.',
+          [1 => $payScaleName]
+        );
+        Session::setStatus($msg, 'Success', 'success');
       }
       else {
-        CRM_Core_Session::setStatus(ts('The Pay Scale for \'%1\' has been added.', array( 1 => $payScale->pay_scale)), 'Success', 'success');
+        $msg = ts(
+          'The Pay Scale for \'%1\' has been added.',
+          [1 => $payScaleName]
+        );
+        Session::setStatus($msg, 'Success', 'success');
       }
 
       $url = CRM_Utils_System::url('civicrm/pay_scale', 'reset=1&action=browse');
-      $session = CRM_Core_Session::singleton();
+      $session = Session::singleton();
       $session->replaceUserContext($url);
     }
   }
+
+  /**
+   * @return array
+   */
+  private function getParams() {
+    $params = $this->exportValues();
+
+    $properties = [
+      'pay_scale',
+      'currency',
+      'amount',
+      'pay_frequency',
+      'is_active'
+    ];
+
+    // set defaults
+    foreach ($properties as $property) {
+      if (!array_key_exists($property, $params)) {
```

Comment:
Should we bother about setting all the other form values to a default if its not present? The formRule already pretty much ensures that all the other fields will be set so this is not necessary. Besides 0 might not be a right default value for some of the fields.
I think what is pretty much needed is just 
`$params['is_active'] = 'CRM_Utils_Array::value('is_active', $params, 0)`;